### PR TITLE
`Development`: Update docker-based monitoring setup

### PR DESCRIPTION
**Pull request by sleiss**
#### General

- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).

### Motivation and Context
We currently have a monitoring-stack consisting of, among others, Grafana and Prometheus.
This PR updates to provided monitoring setup to one that is more similar to the production one.

### Description
- Update versions of Grafana and Prometheus
- Include Dashboards

### Steps for Testing

**Note:** Testing is only possible locally
1. Start Artemis locally, e.g. from your IDE
2. Start the monitoring stack by running `docker-compose -f src/main/docker/monitoring.yml up`
3. Grafana should be reachable at `http://localhost:3000`
4. Login using the username `admin` and password `admin` (You will be asked to change your password now, you can also set it to `admin` again.
5. See that the provided metrics (e.g. Artemis User count work as expected) - You have to open Artemis in your browser so that your connection is counted.

#### Code Review
- [x] Review 1
- [ ] Review 2
#### Manual Tests
- [x] Test 1
- [ ] Test 2

### CODE REVIEW

**Comment by sleiss** [comment]
#removeStale

**Review by 4ludwig4 — COMMENTED** [review]
Sry started already a week ago or longer on this PR review and just didn't know that I have to submit/send/finish the review :rofl: 

Left some comments concerning issues/ideas in the code review which I stumbled onto when working on #5624.

I tried testing it with the new docker-compose setup but couldn't get it working yet.
Didn't yet have time to look into it closer. It's either related to the general network settings or/and the settings of the Prometheus `monitoringIp`. Grafana and Prometheus itself are working fine it seems. Also the dashboard is listed in Grafana.

File: src/main/docker/monitoring.yml [hunk 1]
```diff
@@ -2,11 +2,9 @@
 version: '3.8'
 services:
   artemis-prometheus:
-    image: prom/prometheus:v2.31.1
+    image: prom/prometheus:v2.34.0
     volumes:
-      - ./prometheus/:/etc/prometheus/
-    command:
-      - '--config.file=/etc/prometheus/prometheus.yml'
+      - ./monitoring/prometheus/:/etc/prometheus/
```
[the remark this answers is not in the document]
Comment (sleiss) [reply src/main/docker/monitoring.yml:7]: Indeed, I added a comment regarding this.

File: src/main/docker/monitoring.yml [hunk 2]
```diff
@@ -15,9 +13,9 @@ services:
     # grafana/provisioning/datasources/datasource.yml
     network_mode: 'host' # to test locally running service
   artemis-grafana:
```
[the remark this answers is not in the document]
Comment (sleiss) [reply src/main/docker/monitoring.yml:15]: I renamed the service & container to `grafana-artemis` - this should be fine, right? (and `prometheus-artemis` correspondingly)

**Pull request merged**